local/sdk-sriov/pkg/deviceplugin/server.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deviceplugin
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const (
+	resourceNamePrefix = "networkservicemesh.io/"
+	kubeletNotifyDelay = 30 * time.Second
+	vfioDir            = "/dev/vfio"
+)
+
+// ServerConfig is a struct for configuring device plugin server
+type ServerConfig struct {
+	ResourceName  string
+	ResourceCount int
+	HostBaseDir   string
+	HostPathEnv   string
+}
+
+type devicePluginServer struct {
+	resourceName  string
+	resourceCount int
+	hostBaseDir   string
+	hostPathEnv   string
+	ctx           context.Context
+}
+
+// StartServer creates a new SR-IOV forwarder device plugin server and starts it
+func StartServer(ctx context.Context, config *ServerConfig, manager Manager) error {
+	logFmt := "StartServer: %v"
```

Comment:
1. I think you can use trace.Log(ctx) here for logging.
2. Actually you can use logrus.WithFields(logrus.Fields{}) to avoid this kind formating.

Reply:
It is really great! Replaced it with:
```
logEntry := log.Entry(ctx).WithField("devicePluginServer", "StartServer")
...
logEntry.Info(...)
```